lib/libLLVM_h.jl

```diff
@@ -1,5010 +1,5758 @@
-    @runtime_ccall((:LLVMIsATruncInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsATruncInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAUIToFPInst(Val)
-    @runtime_ccall((:LLVMIsAUIToFPInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAUIToFPInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAZExtInst(Val)
-    @runtime_ccall((:LLVMIsAZExtInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAZExtInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAExtractValueInst(Val)
-    @runtime_ccall((:LLVMIsAExtractValueInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAExtractValueInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsALoadInst(Val)
-    @runtime_ccall((:LLVMIsALoadInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsALoadInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAVAArgInst(Val)
-    @runtime_ccall((:LLVMIsAVAArgInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAVAArgInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAFreezeInst(Val)
-    @runtime_ccall((:LLVMIsAFreezeInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAFreezeInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAAtomicCmpXchgInst(Val)
-    @runtime_ccall((:LLVMIsAAtomicCmpXchgInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAAtomicCmpXchgInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAAtomicRMWInst(Val)
-    @runtime_ccall((:LLVMIsAAtomicRMWInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAAtomicRMWInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAFenceInst(Val)
-    @runtime_ccall((:LLVMIsAFenceInst,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAFenceInst, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAMDNode(Val)
-    @runtime_ccall((:LLVMIsAMDNode,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAMDNode, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMIsAMDString(Val)
-    @runtime_ccall((:LLVMIsAMDString,libllvm[]), LLVMValueRef, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMIsAMDString, libllvm[]) LLVMValueRef (LLVMValueRef,) Val
 end
 
 function LLVMGetValueName(Val)
-    @runtime_ccall((:LLVMGetValueName,libllvm[]), Cstring, (LLVMValueRef,), Val)
+    @runtime_ccall (:LLVMGetValueName, libllvm[]) Cstring (LLVMValueRef,) Val
 end
 
 function LLVMSetValueName(Val, Name)
-    @runtime_ccall((:LLVMSetValueName,libllvm[]), Cvoid, (LLVMValueRef, Cstring), Val, Name)
+    @runtime_ccall (:LLVMSetValueName, libllvm[]) Cvoid (LLVMValueRef, Cstring) Val Name
 end
 
+mutable struct LLVMOpaqueUse end
```

Comment:
Shouldn't these be `struct LLVMOpaqueUse end`?

Reply:
I'm not sure, but [Julia doc](https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/#C-Wrapper-Examples) suggests using `mutable struct`.

Reply:
As opaques are always used in the form of opaque pointers, I guess there is really no difference between a `Ptr{mutable}` and `Ptr{immutable}`. `Ptr`s lives in stack and it also won't affect the layout of struct.

Reply:
As long as its a separate type, and not just an type alias to `Cvoid`, I'm happy.